src/ert/storage/local_ensemble.py

```diff
@@ -291,42 +289,15 @@ def has_data(self) -> List[int]:
         exists : List[int]
             Returns the realization numbers with responses
         """
+
+        ensemble_state = self.get_ensemble_state()
         return [
             i
             for i in range(self.ensemble_size)
-            if all(
-                (self._realization_dir(i) / f"{response}.parquet").exists()
-                or (config.has_finalized_keys and config.keys == [])
-                for response, config in self.experiment.response_configuration.items()
-            )
+            if RealizationStorageState.RESPONSES_LOADED in ensemble_state[i]
         ]
 
-    def realizations_initialized(self, realizations: List[int]) -> bool:
-        """
-        Check if specified realizations are initialized.
-
-        Parameters
-        ----------
-        realizations : list of int
-            List of realization indices.
-
-        Returns
-        -------
-        initialized : bool
-            True if all realizations are initialized.
-        """
-
-        responses = self.get_realization_mask_with_responses()
-        parameters = self.get_realization_mask_with_parameters()
-
-        if len(responses) == 0 and len(parameters) == 0:
-            return False
-
-        return all((responses[real] or parameters[real]) for real in realizations)
-
-    def get_realization_list_with_responses(
-        self, key: Optional[str] = None
-    ) -> List[int]:
+    def get_realization_list_with_responses(self) -> List[int]:
```

Comment:
Not directly related, but do you have a hunch of how much work it would be to just use this function, and not the above one (mask)? Think there is a lot of converting back and forth between the two in different places in the code base.